src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -0,0 +1,633 @@
+                        case SpecialType.System_Int64:
+                        case SpecialType.System_UInt32:
+                        case SpecialType.System_UInt64:
+                        case SpecialType.System_UInt16:
+                        case SpecialType.System_Int16:
+                        case SpecialType.System_SByte:
+                        case SpecialType.System_Byte:
+                        case SpecialType.System_Char:
+                            return RelationalOperationKind.IntegerEqual;
+
+                        case SpecialType.System_Boolean:
+                            return RelationalOperationKind.BooleanEqual;
+
+                        case SpecialType.System_String:
+                            return RelationalOperationKind.StringEqual;
+                    }
+
+                    if (caseValue.Type.TypeKind == TypeKind.Enum)
+                    {
+                        return RelationalOperationKind.EnumEqual;
+                    }
+                }
+
+                return RelationalOperationKind.None;
+            }
+        }
+
+        CaseKind ICaseClause.CaseKind
+        {
+            get { return this.ExpressionOpt != null ? CaseKind.SingleValue : CaseKind.Default; }
+        }
+    }
+
+    partial class BoundTryStatement : ITry
+    {
+        IBlock ITry.Body
+        {
+            get { return this.TryBlock; }
+        }
+
+        ImmutableArray<ICatch> ITry.Catches
+        {
+            get { return this.CatchBlocks.As<ICatch>(); }
+        }
+
+        IBlock ITry.FinallyHandler
+        {
+            get { return this.FinallyBlockOpt; }
+        }
+
+        protected override OperationKind StatementKind
+        {
+            get { return OperationKind.TryStatement; }
+        }
+    }
+
+    partial class BoundCatchBlock : ICatch
+    {
+        IBlock ICatch.Handler
+        {
+            get { return this.Body; }
+        }
+
+        ITypeSymbol ICatch.CaughtType
+        {
+            get { return this.ExceptionTypeOpt; }
+        }
+
+        IExpression ICatch.Filter
+        {
+            get { return this.ExceptionFilterOpt; }
+        }
+
+        ILocalSymbol ICatch.ExceptionLocal
+        {
+            get { return this.LocalOpt; }
+        }
+
+        OperationKind IOperation.Kind
+        {
+            get { return OperationKind.CatchHandler; }
+        }
+
+        SyntaxNode IOperation.Syntax
+        {
+            get { return this.Syntax; }
+        }
+    }
+
+    partial class BoundFixedStatement : IFixed
+    {
+        IVariableDeclaration IFixed.Variables
+        {
+            get { return this.Declarations; }
+        }
+
+        IStatement IFixed.Body
+        {
+            get { return this.Body; }
+        }
+
+        protected override OperationKind StatementKind
+        {
+            get { return OperationKind.FixedStatement; }
+        }
+    }
+
+    partial class BoundUsingStatement: IUsingWithDeclaration, IUsingWithExpression
```

Comment:
Since the class implements both IUsingWith types, callers cannot using type tests to determine which kind of using this is. Given that, consider merging { IUsing, IUsingWithDeclaration, IUsingWithExpression }.